rdkit/Chem/Draw/IPythonConsoleIntegration.py

```diff
@@ -0,0 +1,564 @@
+                    Ids.append((molId,confId))
+                    self.idPaired = Ids
+            
+    @property
+    def selectedModels(self):
+        """ Iterator over all selected models (molecules/conformations) """
+        for (molId, confId) in self.idPaired:
+            yield Chem.MolToMolBlock(self.moldict[molId], confId=confId)
+            
+    @property
+    def selectedMolNames(self):
+        """ Return the names of all selected molecules """
+        return self.rdkit_mol_select
+        
+    @property
+    def selectedConfIds(self):
+        """ Return the names of all selected molecules """
+        return self.rdkit_conf_select
+        
+    @property
+    def selectedMolecules(self):
+        """ Return the selected molecules """
+        return [self.moldict[name] for name in self.selectedMolNames]
+        
+    @property
+    def allConfIds(self):
+        """ Return the number of conformations - use the first selected molecule to determine """
+        nconfIds = self.selectedMolecules[0].GetNumConformers()
+        return list(range(nconfIds))
+        
+    @property
+    def getPropPrecalculated(self):
+        """ Return the precalculated properties """
+        if len(self.selectedMolNames) != 1:
+            return None
+        else:
+            mol = self.moldict[list(self.selectedMolNames)[0]]
+            if len(mol.GetPropNames()) == 0:
+                return 'Not found'
+            else:
+                return {prop: mol.GetProp(prop) for prop in mol.GetPropNames()}
+                
+    
+    
+class MolView3D(object):
+    """This function initiates required widgets and 3Dmol.js viewer
+    uid : an unique id for 3DMol.js viewer, each of the ipywidgets, and all the global objects
+    moldict : it is a molecule dictionary where is keys are string
+    protein : it is rdkit standard mol object
+    molStyle : representation type for moldict or small molecule
+    pStyle : protein representation style
+    ** the rest of the arguments are panels (user may or may not need those)
+    """
+    def __init__(self, 
+                 moldict = None, protein = None,
+                 molStyle = 'stick', pStyle='cartoon',
+                 molPanel = 'full',
+                 stylePanel = False, 
+                 labelPanel = False,
+                 propPanel = False, calculateProperties = False
+                 ):
+        
+        if molPanel not in ('full', 'minimal'):
+            raise KeyError('molPanel can be full or minimal')
+            
+        uid = str(time.time()).replace('.','')
+        
+        self.uid = uid
+        self.molStyle = molStyle
+        self.pStyle = pStyle
+        self.onStart = True
+        self.molPanel = molPanel # molPanel can be full or minimal
+        self.propPanel = propPanel
+        self.stylePanel = stylePanel
+        self.labelPanel = labelPanel
+        
+        wgData = globals().setdefault('wgData', {})
```

Comment:
`wgData` is only used in this class; make it a class field. When you do this, the need for globals disappears and you have 
```
this.wgData = {}
```
Your visualisation (controls and 3D view) is now self-contained and has no other dependency.